pkg/azuredisk/controllerserver.go

```diff
@@ -885,16 +885,66 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 		mc.ObserveOperationWithResult(isOperationSucceeded, consts.SourceResourceID, sourceVolumeID, consts.SnapshotName, snapshotName)
 	}()
 
-	klog.V(2).Infof("begin to create snapshot(%s, incremental: %v) under rg(%s)", snapshotName, incremental, resourceGroup)
-	if rerr := d.cloud.SnapshotsClient.CreateOrUpdate(ctx, subsID, resourceGroup, snapshotName, snapshot); rerr != nil {
-		if strings.Contains(rerr.Error().Error(), "existing disk") {
-			return nil, status.Error(codes.AlreadyExists, fmt.Sprintf("request snapshot(%s) under rg(%s) already exists, but the SourceVolumeId is different, error details: %v", snapshotName, resourceGroup, rerr.Error()))
+	if location != d.cloud.Location {
+		snapshotNameLocal := azureutils.CreateValidDiskName("local_" + snapshotName)
```

Comment:
Instead of duplicating the flow of taking the original snapshot, consider using the same flow to create the local snapshot and doing the copy only if the target region is different.

Reply:
Due to the local and copy snapshotName is the same if we use the same flow to create the local snapshot, then I use snapshotNameLocal instead of snapshotName when target region is different. I think the same name will receive exist error. I consider we can set different snapshotName in the beginning.